src-electron/validation/validation.js

```diff
@@ -144,22 +149,38 @@ function extractIntegerValue(value) {
   if (/^-?\d+$/.test(value)) {
     return parseInt(value)
   } else if (/^[0-9A-F]+$/i.test(value)) {
+    // TODO -- is this supposed to catch hex format?
     return parseInt(value, 16)
   } else {
     return parseInt(value, 16)
   }
 }
-
 function getBoundsInteger(attribute) {
   return {
-    min: extractIntegerValue(attribute.min),
-    max: extractIntegerValue(attribute.max),
+    min: getIntegerValue(attribute.min, attribute.type),
+    max: getIntegerValue(attribute.max, attribute.type),
   }
 }
 
+function getIntegerValue(attribute, type) {
+  let value = extractIntegerValue(attribute)
+  // if its both a signed type and a hex value: checks for the most significant bit , if its set subtract the 16 bit range from it
+  if (
+    isValidHexString(attribute) &&
+    types.isSignedInteger(type) &&
```

Comment:
Use something like getSignAndSizeOfZclType instead since it is not hardcoded for integer type strings which could change based on meta data coming from xml. This function is actually searching the database for the types

Reply:
I don't see why is this functions responsibility to decide this. The whole parent function is only handling integer strings , its called in another function which is checked for integer strings. 

In the case the xml is changed , this part is not even going to run.